test/Microsoft.Dnx.Runtime.Tests/ProjectExtensionsFacts.cs

```diff
@@ -58,6 +58,27 @@ public void GetCompilerOptionsCombinesTargetFrameworkIfNotNull()
             Assert.Equal("x86", options.Platform);
         }
 
+        [Theory]
+        [InlineData("net40", "NET40")]
+        [InlineData(".NETFramework,Version=v4.0", "NET40")]
+        [InlineData(".NETFramework,Version=v4.0,Profile=Client", "NET40_CLIENT")]
+        [InlineData("DOTNET5.1", "DOTNET5_1")]
+        [InlineData(".NETPortable,Version=v4.5,Profile=Profile123", null)]
```

Comment:
Well that was the gist of my question yesterday. We don't currently support `portable-x+y+z` in project.json.